From 541ca3eeede8d5c6d2d38d20b71f5d11531c0a64 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sun, 10 Jan 2021 12:30:20 +0100
Subject: [PATCH 087/179] genirq: clarify naming for out-of-band IPI service

irq_pipeline_send_remote() as a name fails to convey the idea of
sending out-of-band IPIs.

Since this service can only send such type of IRQ, let's rename it to
irq_send_oob_ipi() for the sake of clarity and consistency.

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 arch/arm/kernel/smp.c          | 6 +++---
 arch/arm64/kernel/smp.c        | 6 +++---
 arch/x86/kernel/irq_pipeline.c | 6 +++---
 include/linux/irq_pipeline.h   | 4 ++--
 4 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/arch/arm/kernel/smp.c b/arch/arm/kernel/smp.c
index afac9b4ef..a18fd260d 100644
--- a/arch/arm/kernel/smp.c
+++ b/arch/arm/kernel/smp.c
@@ -763,8 +763,8 @@ static unsigned int get_ipi_count(unsigned int irq, unsigned int cpu)
 	return per_cpu(ipi_counts[irq - ipi_irq_base], cpu);
 }
 
-void irq_pipeline_send_remote(unsigned int irq,
-			      const struct cpumask *cpumask)
+void irq_send_oob_ipi(unsigned int irq,
+		const struct cpumask *cpumask)
 {
 	unsigned int sgi = irq - ipi_irq_base;
 
@@ -776,7 +776,7 @@ void irq_pipeline_send_remote(unsigned int irq,
 	/* Out-of-band IPI (SGI1-2). */
 	__smp_cross_call(cpumask, sgi);
 }
-EXPORT_SYMBOL_GPL(irq_pipeline_send_remote);
+EXPORT_SYMBOL_GPL(irq_send_oob_ipi);
 
 #else
 
diff --git a/arch/arm64/kernel/smp.c b/arch/arm64/kernel/smp.c
index 4e80a3fc6..573954b4d 100644
--- a/arch/arm64/kernel/smp.c
+++ b/arch/arm64/kernel/smp.c
@@ -995,8 +995,8 @@ static unsigned int get_ipi_count(unsigned int irq, unsigned int cpu)
 	return per_cpu(ipi_counts[irq - ipi_irq_base], cpu);
 }
 
-void irq_pipeline_send_remote(unsigned int irq,
-			      const struct cpumask *cpumask)
+void irq_send_oob_ipi(unsigned int irq,
+		const struct cpumask *cpumask)
 {
 	unsigned int sgi = irq - ipi_irq_base;
 
@@ -1008,7 +1008,7 @@ void irq_pipeline_send_remote(unsigned int irq,
 	/* Out-of-band IPI (SGI1-2). */
 	__smp_cross_call(cpumask, sgi);
 }
-EXPORT_SYMBOL_GPL(irq_pipeline_send_remote);
+EXPORT_SYMBOL_GPL(irq_send_oob_ipi);
 
 #else
 
diff --git a/arch/x86/kernel/irq_pipeline.c b/arch/x86/kernel/irq_pipeline.c
index ada4cfbad..22f1df4ef 100644
--- a/arch/x86/kernel/irq_pipeline.c
+++ b/arch/x86/kernel/irq_pipeline.c
@@ -53,12 +53,12 @@ void handle_apic_irq(struct irq_desc *desc)
 	handle_oob_irq(desc);
 }
 
-void irq_pipeline_send_remote(unsigned int ipi,
-			      const struct cpumask *cpumask)
+void irq_send_oob_ipi(unsigned int ipi,
+		const struct cpumask *cpumask)
 {
 	apic->send_IPI_mask_allbutself(cpumask,	apicm_irq_vector(ipi));
 }
-EXPORT_SYMBOL_GPL(irq_pipeline_send_remote);
+EXPORT_SYMBOL_GPL(irq_send_oob_ipi);
 
 void uv_bau_message_interrupt(struct pt_regs *regs);
 
diff --git a/include/linux/irq_pipeline.h b/include/linux/irq_pipeline.h
index fda0a7dd9..157a72c00 100644
--- a/include/linux/irq_pipeline.h
+++ b/include/linux/irq_pipeline.h
@@ -49,8 +49,8 @@ bool handle_oob_irq(struct irq_desc *desc);
 void arch_do_IRQ_pipelined(struct irq_desc *desc);
 
 #ifdef CONFIG_SMP
-void irq_pipeline_send_remote(unsigned int ipi,
-			      const struct cpumask *cpumask);
+void irq_send_oob_ipi(unsigned int ipi,
+		const struct cpumask *cpumask);
 #endif	/* CONFIG_SMP */
 
 void irq_pipeline_oops(void);
-- 
2.38.1

